### ISSUE TITLE
Invalid date values

### ISSUE THREAD

**Issue by macobo** [infused/dbf#85]
Currently, when parsing invalid dates such negative numbers DBF throws an error `invalid date` due to the default behavior of `Date.strptime`.

Is that the desired behavior from behalf of this gem or should this return nil for invalid dates? It seems that DateTime for example is more lenient towards faulty input (https://github.com/infused/dbf/blob/master/lib/dbf/column_type.rb#L71)

Context:

We had a client providing us shape files and due to a bug on their end the shapes started containing negative dates after a point. Even though the columns in question were never used by us, RGeo uses dbf gem to parse every line and this blew up in our face.

We had to hack around this with monkey_patching dbf:
```ruby
DBF::ColumnType::Date.class_eval do
  def type_cast(value)
    value =~ /\d{8}/ && ::Date.strptime(value, '%Y%m%d')
  rescue
    nil
  end
end
```

**Comment by infused** [comment]
Thank you. I just released version 3.1., which includes this patch.

**Comment by macobo** [comment]
Thank you!
